lazer/solana/src/lib.rs

```diff
@@ -149,14 +165,19 @@ pub fn process_update_instruction(
     instruction_args: &[u8],
 ) -> ProgramResult {
     // Verify accounts passed to the instruction.
-    if accounts.len() != 3 {
+    if accounts.len() != 7 {
         return Err(ProgramError::NotEnoughAccountKeys);
     }
-    let sysvar_account = &accounts[0];
+    let payer_account = &accounts[0];
     let data_account = &accounts[1];
-    let pyth_storage_account = &accounts[2];
+    let _pyth_program_account = &accounts[2];
+    let pyth_storage_account = &accounts[3];
+    let pyth_treasury_account = &accounts[4];
+    let system_program_account = &accounts[5];
+    let instructions_sysvar_account = &accounts[6];
```

Comment:
What do all these programs do? Are they actually different accounts entirely?

Reply:
Program accounts are required to allow the contract to call them. This contract calls the pyth contract (`_pyth_program_account `) which calls the system program (`system_program_account`) to make the fee transfer. `pyth_storage_account` and `pyth_treasury_account` are accounts that have to be accessed by the pyth contract. `instructions_sysvar_account` is a special account that allows programs to inspect other instructions within the current transaction, which is needed to validate the signature.